CMakeLists.txt

```diff
@@ -55,6 +61,50 @@ endif()
 # IREE-specific CMake configuration
 #-------------------------------------------------------------------------------
 
+# List of all backends to be built by default:
+set(IREE_ALL_BACKENDS
```

Comment:
The set of compiler backends and drivers isn't necessarily the same. Right now we have the same 3 and they map, but we could have multiple drivers that accept the same compiled executable or multiple compiler paths that generate something that works with the same driver.

We also have slightly different naming for the compiler backends than for the drivers. llvm-ir vs llvm, vulkan-spirv vs vulkan, and vmla just matches. I find this a bit confusing to remember, but it relates to the above and since they're named differently elsewhere I think we should keep that naming consistent. Unless we want to just rename all the compiler backends to match the drivers. I think in that case we should have a plan for what happens when we have two vulkan drivers or two backends that compile to vulkan. As long as we don't dig ourselves into a hole on the naming front.

Maybe the short names could be aliases for the "primary" of each type. So llvm, vulkan, vmla would always work, but then we would have long names for them that are more specific? Anyway this doesn't have to happen as part of your PR, but if we don't change that, I would keep the name lists separate here and stick with the names they already have.

Reply:
Okay, will split this into two separate variables . Regarding the naming, @ScottTodd suggested to rename `Vulkan_SPIRV` to `Vulkan`. But, as disscussed on discord, let's keep this consistent :)